wikimedia/wikipedia-ios
# Add swipe actions to Notifications Center cells

**Phabricator:** https://phabricator.wikimedia.org/T288687

### Notes
This PR adds swipe actions to Notifications Center cells. By left or right swiping (depending on if the device language is LTR/RTL) on a cell, we expose a swipe actions panel with the `More` and `Mark as read/unread` actions. Tapping the `More` button presents an action sheet with additional actions, and tapping `Mark as read/unread` toggles the notification’s read status.

Rather than directly manipulating a cell’s frame, this approach relies on manipulating a cell’s transform to translate its position. Overall, there are some engineering decisions here I’m unhappy about, but so far this implementation has seemed to work well in practice.

### Test Steps
There should never be a case where more than one cell’s swipe panel is open at a time. Any list scrolling, navigation away from the center, tap of edit mode, or change in trait collection should automatically close the panel.

- Set device to LTR language, swipe notifications cell to the left to open swipe action panel. Confirm swipe actions and device rotation works as expected. Close panel.
- Set device to RTL language, swipe notifications cell to the right to open swipe actions panel. Confirm swipe actions and device rotation works as expected. Close panel.
